source/extensions/common/aws/sigv4a_key_derivation.cc

```diff
@@ -38,16 +38,17 @@ EC_KEY* SigV4AKeyDerivation::derivePrivateKey(absl::string_view access_key_id,
     fixed_input.clear();
 
     fixed_input.insert(fixed_input.begin(), {0x00, 0x00, 0x00, 0x01});
-    fixed_input.insert(fixed_input.end(), SigV4ASignatureConstants::SigV4AAlgorithm[0],
-                       SigV4ASignatureConstants::SigV4AAlgorithm[0] +
-                           sizeof(SigV4ASignatureConstants::SigV4AAlgorithm));
+    fixed_input.insert(fixed_input.end(), SigV4ASignatureConstants::SigV4AAlgorithm.data(),
+                       SigV4ASignatureConstants::SigV4AAlgorithm.data() +
+                           SigV4ASignatureConstants::SigV4AAlgorithm.size());
     fixed_input.insert(fixed_input.end(), 0x00);
-    fixed_input.insert(fixed_input.end(), access_key_id.begin(), access_key_id.end());
+    fixed_input.insert(fixed_input.end(), access_key_id.data(),
```

Comment:
OOC what's the added benefit of using pointers instead of iterators (here and below)?
Seems to me the iterators are safer, no?

Reply:
I would prefer the iterators, and they are fine for the non static string views. For a reason I've yet to work out, the static constexpr string view iterators cause this to copy zero bytes with the asan build.

https://github.com/envoyproxy/envoy/pull/34377/files#diff-1cdeea28ecadfd941d26f59f5ea7663f79b11d0c6e553894c8c572fd49afff1bR41

I'm happy to change the non static ones back and leave this one as is, then try and root cause the compilation change.

Reply:
@adisuissa best i've been able to do is determine its an optimisation issue occurring in the asan build to do with static constexpr. The build optimises out all the accesses so it's not even possible to see in the debugger what begin() and end() are returning in this scenario. If you're ok I'll put it back to the way I had this originally, using iterators and just call out why I'm doing the string_view in the comments.